src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -29,7 +39,13 @@
      * @return collection of user, without duplicates
      */
     public List<User> createUsers(final List<String> emails) {
-        return null;
+        List<User> usersList      = emails.stream().
+                                    filter(Objects::nonNull)
+                                    .distinct()
+                                    .filter(email->email.endsWith("@gmail.com"))
```

Comment:
This one is redundant (there is nothing in the requirements about keeping only gmail accounts) but since I didn't cover it in the test, let it be :D